lib/travis/api/v3/models/build.rb

```diff
@@ -80,5 +80,17 @@ def branch_name=(value)
     def clear_debug_options!
       jobs.each { |j| j.update_attribute(:debug_options, nil) }
     end
+
+    def log_complete
+      archived = []
+      jobs.each do |j|
+        archived << Travis::RemoteLog.find_by_job_id(j.id).archived?
+      end
+      if archived.include? false
+        return false
+      else
+        return true
+      end
+    end
```

Comment:
I'd probably write this in a way that it relies on the behaviour of the jobs, e.g.

```ruby
def log_complete
  jobs.all?(&:log_complete)
end
```

But as I mentioned quickly yesterday, this might end up being problematic performance-wise. One request to the logs API per job could get out of hand. Let's see.